sdk/identity/azure-identity/src/client_secret_credential.cpp

```diff
@@ -9,6 +9,40 @@
 #include <chrono>
 #include <sstream>
 
+namespace {
```

Comment:
This isn't wrong to use the unnamed namespace though I'm not sure it has much advantage over using `Azure::Identity::_detail` within the C++ file.  Afterall the unnamed ends up in `::<unique>::....`.  
From what I understand there is some compiler advantage at link time because no symbol resolution occurs. 
 I'm not sure that is worth it over consistency of `_detail` namespace.

Feel free to leave this as its an implementation detail that we could adjust in the future.

Reply:
We should be using unnamed namespaces, unless we need named ones. They were designed for this purpose.
Using unnamed namespaces also allows compiler to make all sorts of right assumptions and make optimizations.
Plus, things from the unnamed namespace won't be made visible by linker.